### PR TITLE
Add additional steps to get coffee files working in tests.

### PR DESCRIPTION
**Pull request by pebcakerror**
Issue #476 

I figured out how to get coffeescript working with in test. 

I'm no gulp expert so this may not be the _best_ way. Someone with more gulp knowledge should definitely git this a look to be sure.

### CODE REVIEW

File: docs/recipes/coffeescript.md [hunk 1]
```diff
@@ -23,10 +26,10 @@ gulp.task('scripts', () => {
 });
 ```
 
-### 3. Add `scripts` as a dependency of `html` and `serve`
+#### 2.2 Add `scripts` as a dependency of `html`, `serve`, `server:test`, and `build`
 
 ```js
-gulp.task('html', ['styles', 'scripts'], () => () {
```
Comment (eddiemonge) [inline docs/recipes/coffeescript.md:29]: this looks like a typo and should be fixed regardless

Comment (pebcakerror) [inline docs/recipes/coffeescript.md:29]: This line corrects the typo from the original recipe.

**Comment by pebcakerror** [comment]
Is there anything I can do to help get this PR accepted?

File: docs/recipes/coffeescript.md [hunk 2]
```diff
@@ -5,15 +5,18 @@ With this setup you can freely mix `.js` and `.coffee` files in your `app/script
...
 
+
 ### 1. Install the [gulp-coffee](https://github.com/wearefractal/gulp-coffee) plugin
 
 ```
 $ npm install --save-dev gulp-coffee
 ```
 
-### 2. Create a `scripts` task
+### 2. app/scripts:
```
Comment (silvenon) [inline docs/recipes/coffeescript.md:15]: Probably better to call this "Source" instead of "app/scripts".

**Comment by silvenon** [comment]
While we're at it, could you please change this:

``` js
gulp.watch('app/scripts/**/*.coffee', ['scripts', reload]);
```

into this:

``` js
gulp.watch('app/scripts/**/*.coffee', ['scripts']);
```

then remove this line:

``` js
'.tmp/scripts/**/*.js',
```

from the `serve` and `serve:test` tasks. This removes excessive reloading, we want to reload just once, when all files are compiled, so please make sure this at the end of the `scripts` and `scripts:test` tasks:

```
.pipe(reload({stream: true}));
```

Please make sure to add correct diff symbols (`-` and `+`).

**Comment by silvenon** [comment]
I just re-read the title of your PR. You don't have to do all that, just what I commented on the diff.

**Comment by ivan-vilches** [comment]
I am happy starting with this generator, if its possible update the coffescript recipe wil be nice . Am not gulp guru . Thanks

**Comment by sindresorhus** [comment]
@silvenon What's holding this back from being merged?

**Comment by silvenon** [comment]
@inspiredbytes thanks for your contribution! 👍 

@sindresorhus @ivan-vilches this recipe needs other updates, it has some mistakes unrelated to the PR. I'll post here in an hour when I'm done.

**Comment by silvenon** [comment]
Actually, tomorrow, there's more work than I thought.

**Comment by silvenon** [comment]
@inspiredbytes @ivan-vilches I thoroughly updated [the recipe](https://github.com/yeoman/generator-webapp/blob/master/docs/recipes/coffeescript.md). Now it should work just fine with the [latest version of generator-webapp](https://github.com/yeoman/generator-webapp/releases/tag/v2.3.0).

**Pull request merged**